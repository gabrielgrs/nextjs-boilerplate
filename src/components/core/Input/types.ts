import type { InputHTMLAttributes } from 'react'

export type Input = InputHTMLAttributes<HTMLInputElement> & { invalid?: boolean }
