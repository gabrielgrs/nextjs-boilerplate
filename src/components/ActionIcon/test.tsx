import { render, screen, fireEvent } from 'utils/test'
import ActionIcon from './index'

describe('<ActionIcon />', () => {
  it('should match snapshot', () => {
    const { container } = render(<ActionIcon backgroundColor="primary">Action Icon</ActionIcon>)
    expect(container).toMatchSnapshot()
  })

  it('should render right color', () => {
    render(<ActionIcon backgroundColor="primary">Action Icon</ActionIcon>)

    expect(screen.getByText(/action icon/i)).toHaveStyle({
      backgroundColor: '#0bb58d',
    })
  })

  it('should dispatch on click on pressed', () => {
    const onClick = jest.fn()
    render(
      <ActionIcon backgroundColor="primary" onClick={onClick}>
        Action Icon
      </ActionIcon>
    )

    fireEvent.click(screen.getByText(/action icon/i))

    expect(onClick).toHaveBeenCalled()
  })
})
