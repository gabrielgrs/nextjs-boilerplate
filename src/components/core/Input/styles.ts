import styled, { css } from 'styled-components'

export const Input = styled.input<{ invalid?: boolean }>`
  ${({ theme, invalid }) => css`
    height: 32px;
    width: 100%;
    border: none;
    background: transparent;
    border-bottom: solid ${invalid ? theme.colors.danger : theme.colors.silver} 1px;
    transition: 0.4s;
    color: ${theme.colors.black};

    &:focus {
      border: none;
      outline: none;
      border-bottom: solid ${theme.colors.primary} 1px;
    }
  `}
`
