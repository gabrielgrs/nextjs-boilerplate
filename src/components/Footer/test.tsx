import { render, screen } from 'utils/test'
import Footer from './index'

describe('<Footer />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Footer />)
    expect(container).toMatchSnapshot()
  })

  it('should render link', () => {
    render(<Footer />)

    expect(screen.getByRole('link', { name: /gabrielgrs/i })).toBeInTheDocument()
  })
})
