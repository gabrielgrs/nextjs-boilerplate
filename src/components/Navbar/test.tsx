import { render } from 'utils/test'
import Navbar from './index'

describe('<Navbar />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Navbar />)
    expect(container).toMatchSnapshot()
  })
})
