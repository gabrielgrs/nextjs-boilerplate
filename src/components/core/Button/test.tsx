import { render, screen } from 'utils/test'
import Button from './index'

describe('<Button />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Button>Button Content</Button>)

    expect(container).toMatchSnapshot()
  })

  it('should render button', () => {
    render(<Button>Button Content</Button>)

    const element = screen.getByRole('button', { name: /Button Content/i })
    expect(element).toBeInTheDocument()
  })

  it('should render a fullWidth version', () => {
    render(<Button fullWidth>Button Content</Button>)

    expect(screen.getByRole('button', { name: /Button Content/i })).toHaveStyle({
      width: '100%',
    })
  })

  it('should render a disabled Button', () => {
    render(<Button disabled>Button Content</Button>)

    expect(screen.getByRole('button', { name: /Button Content/i })).toHaveStyleRule(
      'cursor',
      'not-allowed',
      {
        modifier: ':disabled',
      }
    )
  })

  it('should render Button with variant', () => {
    render(<Button variant="danger">Button Content</Button>)
    const element = screen.getByRole('button', { name: /Button Content/i })
    expect(element).toHaveStyle({
      backgroundColor: '#F5726C',
    })
  })
})
