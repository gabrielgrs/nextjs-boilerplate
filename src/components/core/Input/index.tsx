import { forwardRef } from 'react'
import * as S from './styles'
import type * as T from './types'

function Input(props: T.Input, ref: any) {
  return <S.Input ref={ref} {...props} />
}

export default forwardRef(Input)
