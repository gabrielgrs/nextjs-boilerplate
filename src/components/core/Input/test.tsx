import userEvent from '@testing-library/user-event'
import { render, screen, fireEvent } from 'utils/test'
import Input from './index'

describe('<Input />', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Input />)

    expect(container).toMatchSnapshot()
  })

  it('Renders with placeholder', () => {
    render(<Input placeholder="Placeholder" />)

    expect(screen.getByPlaceholderText('Placeholder')).toBeInTheDocument()
  })

  it('Changes its value when typing', () => {
    const onChange = jest.fn()
    render(<Input onChange={onChange} placeholder="type something" name="Input" />)
    const input = screen.getByPlaceholderText('type something')

    const typedText = 'new text'
    fireEvent.change(input, { target: { value: typedText } })

    expect(input).toHaveValue(typedText)
  })

  it('Should be disabled', () => {
    const onChange = jest.fn()
    render(<Input disabled onChange={onChange} placeholder="type something" name="Input" />)
    const input = screen.getByPlaceholderText('type something')

    const typedText = 'new text'
    fireEvent.change(input, { target: { value: typedText } })

    expect(input).toBeDisabled()
  })

  it('Renders with error', () => {
    render(<Input invalid placeholder="type something" name="Input" />)

    expect(screen.getByPlaceholderText('type something')).toHaveStyle({
      borderBottom: 'solid 1px #F5726C',
    })
  })

  it('Is accessible by tab', () => {
    render(<Input placeholder="type something" />)

    const input = screen.getByPlaceholderText('type something')
    expect(document.body).toHaveFocus()

    userEvent.tab()
    expect(input).toHaveFocus()
  })
})
