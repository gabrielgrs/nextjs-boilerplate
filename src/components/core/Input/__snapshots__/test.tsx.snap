// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should match snapshot 1`] = `
.c0 {
  height: 32px;
  width: 100%;
  border: none;
  background: transparent;
  border-bottom: solid #ccc 1px;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  color: #282828;
}

.c0:focus {
  border: none;
  outline: none;
  border-bottom: solid #0bb58d 1px;
}

<div>
  <input
    class="c0"
  />
</div>
`;
